proto/p4/config/p4info.proto

```diff
@@ -43,24 +41,24 @@ message Preamble {
   // (e.g. direct resource list for a table) without ambiguity. Note that id 0
   // is reserved and means "invalid id".
   int32 id = 1;
-  // fully qualified name of the P4 object
-  string name = 2;
-  repeated string annotations = 3;
-}
-
-message HeaderField {
-  Preamble preamble = 1;
-  int32 bitwidth = 2;
-}
-
-message HeaderFieldList {
-  Preamble preamble = 1;
-  repeated int32 header_field_ids = 2;
+  // fully qualified name of the P4 object, e.g. c1.c2.ipv4_lpm
+  string path = 2;
+  // an alias for the P4 object, probably shorter than its path. The only
+  // constraint is for it to be unique with respect to other P4 objects of the
```

Comment:
That would disallow using `PuntTable` twice for 2 different tables. Bu the names could be `c2.PuntTable` and `c1.c3.PuntTable`, or `PuntTable1` and `PuntTable2` (if overridden by programmer).
My understanding is that for your use case, you would be using the `path` anyway (and organize namespaces accordingly) and would not use `name`.